src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,185 @@
+
+import { AlertVariant } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Text, TextContent } from "@patternfly/react-core/dist/esm/components/Text";
+
+import cockpit from 'cockpit';
+import { FileInfo } from 'cockpit/fsinfo.ts';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { superuser } from 'superuser';
+
+import { useFilesContext } from '../app.tsx';
+import type { ClipboardInfo } from '../app.tsx';
+import { get_owner_candidates } from '../ownership.tsx';
+
+const _ = cockpit.gettext;
+
+async function pasteAsOwner(clipboard: ClipboardInfo, dstPath: string, owner: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
+    try {
+        await cockpit.spawn([
+            "cp",
+            "--recursive",
+            ...clipboard.files.map(file => clipboard.path + "/" + file.name),
+            dstPath
+        ], { superuser: "try" });
+
+        await cockpit.spawn([
+            "chown",
+            "--recursive",
+            owner,
+            ...clipboard.files.map(file => dstPath + "/" + file.name),
+        ], { superuser: "try" });
+    } catch (err) {
+        const e = err as cockpit.BasicError;
+        addAlert(e.message, AlertVariant.danger, `${new Date().getTime()}`);
+
+        // cleanup potentially copied files in case of "chown" fail
+        try {
+            await cockpit.spawn([
+                "rm",
+                "-rf",
+                ...clipboard.files.map(file => dstPath + "/" + file.name)
+            ], { superuser: "try" });
+        } catch (ex) {
+            console.warn(`Failed to clean up copied files in ${dstPath}`, ex);
+        }
+    }
+}
+
+function makeCandidatesMap(currentUser: cockpit.UserInfo, cwdInfo: FileInfo, clipboard: ClipboardInfo) {
+    const map: Map<string, string> = new Map();
+    const candidates = get_owner_candidates(currentUser, cwdInfo);
+
+    // also add current ownership if it is same for all files (shallow check)
+    const firstFile = clipboard.files[0];
+    const firstOwnerStr = `${firstFile.user}:${firstFile.group}`;
+    if (clipboard.files.every(file => file.user === firstFile.user && file.group === firstFile.group)) {
+        candidates.add(firstOwnerStr);
+    }
+
+    candidates.forEach(owner => {
+        const split = owner.split(':');
+        let key = owner;
+        if (split.length === 2 && split[0] === split[1]) {
+            key = split[0];
+        }
+        if (owner === firstOwnerStr) {
+            key = `${key} ${_("(original owner)")}`;
+        }
+
+        map.set(key, owner);
```

Comment:
There is a potential duplication here if an owner appears both in get_owner_candidates and firstOwner. I suppose that's what you meant yesterday with "stable map insertion order", but it still feels a bit brittle to rely on that. Please at least add a comment that explains this.